mdiary.c:156:3: warning: '__builtin___memset_chk' will always overflow destination buffer [-Wbuiltin-memcpy-chk-size]
  memset(&d, '\0', sizeof(dinfo));
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/secure/_string.h:76:3: note: expanded from macro 'memset'
                __builtin___memset_chk (dest, __VA_ARGS__, __darwin_obsz0 (dest))
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
--- src/mdiary.c.orig	2018-11-05 07:32:08.000000000 -0600
+++ src/mdiary.c	2018-11-05 07:32:34.000000000 -0600
@@ -153,7 +153,7 @@
   munmap(d->text, d->length);
   close(d->fd);
 
-  memset(&d, '\0', sizeof(dinfo));
+  memset(d, '\0', sizeof(dinfo));
 } 
 
 /* mdiary_note - returns a specific note in the mdiary */
